src/components/chart/Chart.vue

```diff
@@ -32,8 +32,16 @@ export default {
         }
     },
     watch: {
-        data() {
-            this.reinit();
+        /*
+         * Use deep watch to enable triggering watch for changes within structure
+         * otherwise the entire data object needs to be replaced to trigger watch
+         */
+        data: {
+            handler(val, oldVal) {
+                console.log('watch data')
+                this.reinit();
+            },
```

Comment:
I'm not in any way official, but if you wanted to get this merged in, I'd recommend this small change:

```suggestion
            handler() {
                this.reinit();
            },
```